### PR TITLE
Payara-2154 jdk9

### PR DESCRIPTION
**Pull request by arjantijms**
Payara-2154 several updates for JDK9 compatibility

### CODE REVIEW

**Comment by arjantijms** [comment]
Jenkins quick test Payara 5 please

**Comment by arjantijms** [comment]
Jenkins quick test Payara 5 please

**Comment by arjantijms** [comment]
Jenkins quick test Payara 5 please

File: appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java [hunk 1]
```diff
@@ -66,123 +67,69 @@
...
+     * what we intend to happen. Initializes and updates the Vector of 
+     * jars that have been successfully closed.
+     *
+     * <p>
+     * Any errors are logged.
+     *
+     * @param classLoader the instance of URLClassLoader (or a subclass) 
+     * @param jarsClosed a Vector of Strings that will contain the names of jars 
+     *        successfully closed; can be null if the caller does not need the information returned
+     *
```
Comment (Pandrex247) [inline appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java]: Pretty sure our code style is no random empty lines between the tags in javadoc
(ﾉ≧∇≦)ﾉ ﾐ ┸━┸

File: appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java [hunk 2]
```diff
@@ -300,6 +251,89 @@ public static void releaseLoader(URLClassLoader classLoader) {
     }
     
     /**
+     *Initializes the class.
+     *<p>
+     *Each utility method should invoke init() before doing their own work
```
Comment (Pandrex247) [inline appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java]: Should be spaces between the * and text - like you did above.

File: appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardWrapper.java [hunk 3]
```diff
@@ -1359,6 +1386,9 @@ private synchronized Servlet loadServlet() throws ServletException {
         return servlet;
     }
 
+    private String msg(String key, Object... arguments) {
```
Comment (Pandrex247) [inline appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardWrapper.java]: That isn't a proper method name :P

File: appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardWrapper.java [hunk 4]
```diff
@@ -1449,17 +1478,17 @@ public Class run() throws Exception{
...
+                e);
+            
+            throw new ServletException(format(rb.getString(CANNOT_FIND_SERVLET_CLASS_EXCEPTION), actualClass), e);
         }
 
         if (clazz == null) {
-            String msg = MessageFormat.format(rb.getString(LogFacade.CANNOT_FIND_SERVLET_CLASS_EXCEPTION), actualClass);
             unavailable(null);
-            throw new ServletException(msg);
+            throw new ServletException(format(rb.getString(CANNOT_FIND_SERVLET_CLASS_EXCEPTION), actualClass));
```
Comment (Pandrex247) [inline appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardWrapper.java]: Could you not use the `msg` method?

File: appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardWrapper.java [hunk 5]
```diff
@@ -1475,21 +1504,20 @@ public Class run() throws Exception{
      * Initializes the given servlet instance, by calling its init method.
      */
     private void initServlet(Servlet servlet) throws ServletException {
+
```
Comment (Pandrex247) [inline appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardWrapper.java]: (┛✧Д✧))┛彡┻━┻

File: appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java [hunk 6]
```diff
@@ -66,123 +67,69 @@
...
  */
 public class ClassLoaderUtil {
-
-    /** records whether initialization has been completed */
-    private static boolean isInitialized = false;
     
-    /** names of classes and fields of interest for closing the loader's jar files */
-    private static final String URLCLASSLOADER_UCP_FIELD_NAME = "ucp";
+    /** Names of classes and fields of interest for closing the loader's jar files */
+
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java]: It's actually not random, but an intended empty line :P It's to separate the banner from the entire block of constants and the single contant. A comment banner attached to a single constant would most often been seen as commenting that constant only and not all others.

But actually, the style here should be ``// ###`` or something like it, as ``/**`` is more or less intended to comment only that to which it is "attached" ;)

File: appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java (hunk 1 above)
Comment (arjantijms) [reply appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java]: It's actually to separate the `param` ones from the `return` one ;)

File: appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java (hunk 2 above)
Comment (arjantijms) [reply appserver/common/glassfish-ee-api/src/main/java/com/sun/appserv/ClassLoaderUtil.java]: Yeah indeed, I did some base formatting but seemingly didn't do all of it

**Comment by arjantijms** [comment]
Jenkins quick test Payara 5 please

File: appserver/admin/payara_template/src/main/resources/config/domain.xml [hunk 7]
```diff
@@ -166,7 +166,12 @@
         </message-security-config>
 	<property value="SHA-256" name="default-digest-algorithm" />
       </security-service>
+      
       <java-config classpath-suffix="" system-classpath="" debug-options="-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=%%%JAVA_DEBUGGER_PORT%%%">
+        <jvm-options>-XX:+IgnoreUnrecognizedVMOptions</jvm-options>
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply appserver/admin/payara_template/src/main/resources/config/domain.xml]: I think you're right

**Comment by arjantijms** [comment]
Jenkins quick test Payara 5 please

**Comment by arjantijms** [comment]
Jenkins quick test Payara 5 please

**Comment by arjantijms** [comment]
Jenkins quick test Payara 5 please

**Pull request merged**